### PR TITLE
feat: Speakers Call Detail for an Event

### LINKED ISSUE #927: View Detailed Speakers Call for an Event
**Is it a ?**

- [ ] Bug Report

- [x] Feature Request
    
- [ ] Chore

**Feature Description**

The user should be able to view the speaker call for an event.

**Parent Issue**
#731 
   

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #927 

Changes: 
1. Request Speaker Call detail from server.
2. Save Speaker Call object in database.
3. Added Migration for SpeakerCall Table.

Not ready for review yet.

Screenshots for the change: 
speaker

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+@Data
+@Builder
+@Type("speakers-call")
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString(exclude = "event")
+@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy.class)
+@Table(database = OrgaDatabase.class, allFields = true)
+@EqualsAndHashCode()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Unneeded

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public String announcement;
+    public String hash;
+    public String privacy;
+
+    @JsonSerialize(using = ObservableStringSerializer.class)
+    @JsonDeserialize(using = ObservableStringDeserializer.class)
+    public ObservableString startsAt = new ObservableString();
+    @JsonSerialize(using = ObservableStringSerializer.class)
+    @JsonDeserialize(using = ObservableStringDeserializer.class)
+    public ObservableString endsAt = new ObservableString();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Use standard Date

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+@Data
+@Builder
+@Type("speakers-call")
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString(exclude = "event")
+@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy.class)
+@Table(database = OrgaDatabase.class, allFields = true)
+@EqualsAndHashCode()
+@SuppressWarnings("PMD.TooManyFields")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Unneeded

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@Builder
+@Type("speakers-call")
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Unneeded

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@Builder
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Unneeded

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+@Type("speakers-call")
+@AllArgsConstructor
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Unneeded

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCallRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    public Observable<SpeakersCall> getSpeakersCall(long eventId, boolean reload) {
+        Observable<SpeakersCall> diskObservable = Observable.defer(() ->
+            repository.getItems(SpeakersCall.class, SpeakersCall_Table.event_id.eq(eventId))
+        );
+
+        Observable<SpeakersCall> networkObservable = Observable.defer(() ->
+            speakersCallApi.getSpeakersCall(eventId)
+                .doOnNext(speakersCall -> repository
+                        .save(SpeakersCall.class, speakersCall)
+                        .subscribe()));
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCallRepositoryImpl.java]: Use syncsave

Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCallRepositoryImpl.java]: `syncSave` is for saving list of items in db but the `GET /v1/events/{event_identifier}/speakers-call` endpoint returns only the most recent speakersCall object. So I think `save` will work fine.

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java (hunk 2 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: Sorry, I didn't get what do you mean by standard Date ?

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCallRepositoryImpl.java (hunk 7 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCallRepositoryImpl.java]: @iamareebjamal .

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java (hunk 2 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: @iamareebjamal .

**Comment by iamareebjamal** [comment]
There are some UI concerns. Other than that, looks fine

**Comment by Masquerade0097** [comment]
@iamareebjamal What kind of improvement can I do for UI?

**Comment by iamareebjamal** [comment]
The text is too large and the communication about date and all are very generic. I know this is derivative of the work in Event Detail but I'm the culprit there for bad UI. Take a look at some nice apps like recently launched Google Tasks to make it a good user experience for users. It can be improved and new ideas are appreciated which can be applied for the whole app

Also, I'm not sure hash/public private is relevant but this should be taken up with server guys as to what they mean

File: app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/data/speakerscall/SpeakersCall.java]: I meant LocalDate

**Comment by Masquerade0097** [comment]
@iamareebjamal What about this design ? It isn't polished yet, is a prototype.
1 - screen 1

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

**Comment by Masquerade0097** [comment]
@iamareebjamal @sridharjajoo @mishuvs The Build was completing successfully before but after merging it's failing. Any idea why ?

**Comment by Masquerade0097** [comment]
@iamareebjamal The build was working before merging this branch with development on my local setup. After merging I'm getting this error 
`org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:compileDebugJavaWithJavac'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:52)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:37)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:98)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:51)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:37)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:35)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:25)
	at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:156)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:141)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:114)
	at com.android.build.gradle.tasks.factory.AndroidJavaCompile.compile(AndroidJavaCompile.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
	... 33 more
`

On downgrading `tools.build:gradle` from 3.1.1 to 3.0.0, the build is working again.

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
Elevation is too much, reduce it

**Comment by iamareebjamal** [comment]
Also, increase the padding of the icon

**Comment by Masquerade0097** [comment]
Done with the changes.

**Comment by iamareebjamal** [comment]
@Masquerade0097 Please resolve merge conflicts

**Pull request merged**